lldb/include/lldb/lldb-enumerations.h

```diff
@@ -1362,6 +1362,12 @@ enum Severity {
   eSeverityInfo, // Equivalent to Remark used in clang.
 };
 
+enum NotificationType {
```

Comment:
Might be good to put a comment here saying (a) this is used for SBNotificationCallbacks and (b) this is where you should add any other notification callbacks (i.e. target creation and destruction...)  That should be made clear so we don't invent another mechanism for this elsewhere...